### PR TITLE
feat(client): add isCompatible helper

### PR DESCRIPTION
**Pull request by voliva**

### CODE REVIEW

File: packages/client/src/compatibility.ts [hunk 1]
```diff
@@ -223,28 +229,65 @@ export const minCompatLevel = (levels: {
...
+  (...args: [...A, runtime: CompatibilityToken<D>]): T
 }
 
 export interface CompatibilityFunctions<D> {
   /**
    * Returns the `CompatibilityLevel` for this call comparing the descriptors
    * generated on dev time with the current live metadata.
    */
-  getCompatibilityLevel: WithOptionalRuntime<CompatibilityLevel, D>
+  getCompatibilityLevel(): Promise<CompatibilityLevel>
```
Comment (voliva) [inline packages/client/src/compatibility.ts:260]: Even though `: WithOptionalRuntime<...>` would work, this is part of the public API and the docs the IDE shows are of the `WithOptionalRuntime` overloads.

So I've decided to inline them instead

Comment (carlosala) [reply packages/client/src/compatibility.ts:260]: I was about to say it is a great idea to inline them. Nice!

File: packages/client/src/compatibility.ts [hunk 2]
```diff
@@ -223,28 +229,65 @@ export const minCompatLevel = (levels: {
...
+  /**
+   * Returns the `CompatibilityLevel` for this call comparing the descriptors
+   * generated on dev time with the current live metadata.
+   *
+   * @param compatibilityToken  CompatibilityToken awaited from
+   *                            typedApi.compatibilityToken.
+   */
+  getCompatibilityLevel(
+    compatibilityToken: CompatibilityToken<D>,
+  ): Promise<CompatibilityLevel>
```
Comment (carlosala) [inline packages/client/src/compatibility.ts:270]: This should be sync, right?

**Review by carlosala — APPROVED** [review]
LGTM! 🚀

**Pull request merged**